sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -337,37 +367,36 @@ def _incr_mean_variance_axis0(np.ndarray[floating, ndim=1] X_data,
         np.ndarray[floating, ndim=1] new_n
         np.ndarray[floating, ndim=1] updated_n
         np.ndarray[floating, ndim=1] last_over_new_n
-        np.ndarray[floating, ndim=1] counts_nan
 
     # Obtain new stats first
-    new_n = np.full(n_features, n_samples, dtype=dtype)
+    new_n = np.empty(shape=n_features, dtype=dtype)
```

Comment:
We don't need to pre-allocate `new_n` anymore now that's it's returned by `_csr_mean_variance_axis0`.